controllers/sandbox_controller.go

```diff
@@ -280,6 +285,40 @@ func (r *SandboxReconciler) reconcilePod(ctx context.Context, sandbox *sandboxv1
 	return pod, nil
 }
 
+// reconcileSandboxExpiry will check if a sandbox has expired, and if so, delete it.
+// If the sandbox has not expired, it will requeue the request for the remaining time.
+
+func (r *SandboxReconciler) reconcileSandboxExpiry(ctx context.Context, sandbox *sandboxv1alpha1.Sandbox) (ctrl.Result, error) {
+	log := log.FromContext(ctx)
+
+	var expiryTime time.Time
+	if sandbox.Spec.ShutdownTime == nil {
+		log.Info("Sandbox ShutdownTime is not set, skipping.")
+		return reconcile.Result{}, nil
+	}
+
+	expiryTime = sandbox.Spec.ShutdownTime.Time
+
+	// Calculate remaining time
+	remainingTime := time.Until(expiryTime)
+	if remainingTime <= 0 {
+		log.Info("Sandbox has expired, deleting")
+		if err := r.Delete(ctx, sandbox); err != nil {
```

Comment:
Ah - I'm really not sure about this.  Do we want to free resources but keep the Sandbox?

Reply:
can do. not sure what the requirement is. 
who cleans up the sandbox ?